Forms/Fields/DateOfBirthDropdownField.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DUCK.Forms.Fields
+{
+	public class DateOfBirthDropdownField : AbstractFormField
+	{
+		[SerializeField]
+		private Dropdown dayDropdown;
+		[SerializeField]
+		private Dropdown monthDropdown;
+		[SerializeField]
+		private Dropdown yearDropdown;
+
+		[SerializeField]
+		private Color onValueChangedColor;
+
+		private int defaultDay;
+		private int defaultMonth;
+		private int defaultYear;
+
+		protected override void Awake()
+		{
+			base.Awake();
+
+			defaultDay = dayDropdown.value;
+			defaultMonth = monthDropdown.value;
+			defaultYear = yearDropdown.value;
+
+			dayDropdown.onValueChanged.AddListener(value => { dayDropdown.captionText.color = onValueChangedColor; });
+			monthDropdown.onValueChanged.AddListener(value =>
+			{
+				monthDropdown.captionText.color = onValueChangedColor;
+				HandleValueChanged(value);
+			});
+			yearDropdown.onValueChanged.AddListener(value =>
+			{
+				yearDropdown.captionText.color = onValueChangedColor;
+				HandleValueChanged(value);
+			});
+
+			monthDropdown.AddOptions(DateOfBirthUtils.Months.ToList());
+			dayDropdown.AddOptions(DateOfBirthUtils.Days.Select(day => day.ToString()).ToList());
+			yearDropdown.AddOptions(DateOfBirthUtils.Years.Select(year => year.ToString()).ToList());
+
+			HandleValueChanged(monthDropdown.value);
+		}
+
+		public override object GetValue()
+		{
+			return new DateOfBirth(GetDay(), GetMonth(), GetYear());
+		}
+
+		private void HandleValueChanged(int value)
+		{
+			var dayValue = dayDropdown.value;
+			dayDropdown.ClearOptions();
+			var daysInMonth = DateTime.DaysInMonth(GetYear(), GetMonth());
+			var days = new int[daysInMonth];
+			for (var day = 0; day < daysInMonth; day++)
+			{
+				days[day] = day + 1;
+			}
+			dayDropdown.AddOptions(days.Select(day => day.ToString()).ToList());
+			if (daysInMonth > dayValue)
+			{
+				dayDropdown.value = dayValue;
+			}
+		}
+
+		private int GetYear()
+		{
+			return int.Parse(yearDropdown.captionText.text);
```

Comment:
Why not use a set of int values to represent the year - why do years alone get parsed from strings?

Reply:
It comes from a Text component, Unity's dropdown fields work this way. Year is not alone, Days work this way too.